src/BinSkim.Rules/PERules/BA2006.BuildWithSecureTools.cs

```diff
@@ -299,93 +241,124 @@
                         string.Join(", ", sorted)));
         }
 
-        private string BuildCompilerIdentifier(ObjectModuleDetails omDetails)
-        {
-            return omDetails.CompilerName + ":" + omDetails.Language + ":" + omDetails.CompilerBackEndVersion;
-        }
-
-        private string BuildMinimumCompilersList(BinaryAnalyzerContext context, Dictionary<Language, List<ObjectModuleDetails>> languageToBadModules)
+        internal void GenerateMessageParametersAndLog(BinaryAnalyzerContext context, Dictionary<Language, List<ObjectModuleDetails>> languageToBadModules)
         {
+            var sb = new StringBuilder();
```

Comment:
>StringBuilder [](http://example.com/codeflow?start=25&length=13)

Can we use a per-instance member variable for this? Our threading model will permit it, won't it?

Reply:
yes.
made the change.
thanks